### PR TITLE
[BottomAppBar] Add conformance to MDCElevatable, MDCElevationOverriding

### LINKED ISSUE #8027: [BottomAppBar] Internal issue: b/138203607
This was filed as an internal issue. If you are a Googler, please visit [b/138203607](http://b/138203607) for more details.

---
#### Internal data
- Associated internal bug: [b/138203607](http://b/138203607)

### PR DESCRIPTION
**Pull request by bryanoltman**
Adds Bottom App Bar conformance to `MDCElevatable` `MDCElevationOverriding`. Note that the bottom app bar view's shadow is not using `MDCShapedShadowLayer`. I explored this in #8092, but the work required to adapt `MDCBottomAppBarLayer` seemed out-of-scope.

Closes #8027

### CODE REVIEW

File: components/BottomAppBar/src/MDCBottomAppBarView.h [hunk 1]
```diff
@@ -137,4 +137,22 @@ typedef NS_ENUM(NSInteger, MDCBottomAppBarFloatingButtonPosition) {
...
+ hierarchy ancestors.
+
+ Use this block to respond to elevation changes in the view or its ancestor views.
+
+ @param elevation The @c mdc_currentElevation plus the @c mdc_currentElevation of all ancestor
+ views.
+ @param object This bottom app bar.
+ */
+@property(nonatomic, copy, nullable) void (^mdc_elevationDidChangeBlock)
+    (MDCBottomAppBarView *_Nonnull object, CGFloat elevation);
```
Comment (codeman7) [inline components/BottomAppBar/src/MDCBottomAppBarView.h]: _Tip_: Change `object` to `bottomAppBar`

File: components/BottomAppBar/src/MDCBottomAppBarView.h [hunk 2]
```diff
@@ -137,4 +137,22 @@ typedef NS_ENUM(NSInteger, MDCBottomAppBarFloatingButtonPosition) {
     (MDCBottomAppBarView *_Nonnull bottomAppBar,
      UITraitCollection *_Nullable previousTraitCollection);
 
+/**
+ The elevation of the bottom app bar. Defaults to @c MDCShadowElevationBottomNavigationBar.
```
Comment (codeman7) [inline components/BottomAppBar/src/MDCBottomAppBarView.h]: Can we add MDCShadowElevationBottomAppBar and make that the default?

**Review by codeman7 — APPROVED** [review]
Please update the shadow elevation value before merging.

File: components/BottomAppBar/src/MDCBottomAppBarView.m [hunk 3]
```diff
@@ -86,6 +88,8 @@ - (void)commonMDCBottomAppBarViewInit {
 
   self.barTintColor = UIColor.whiteColor;
   self.shadowColor = UIColor.blackColor;
+  self.elevation = MDCShadowElevationBottomNavigationBar;
```
Comment (romoore) [inline components/BottomAppBar/src/MDCBottomAppBarView.m]: Please assign the elevation to the ivar directly. Accessing `self` in an initializer is discouraged.  In this case, `setElevation:` calls out to the category method and triggers a bunch of other code.

```objc
_elevation = MDCShadowElevationBottomNavigationBar;
```

File: components/BottomAppBar/src/MDCBottomAppBarView.m [hunk 4]
```diff
@@ -302,6 +306,14 @@ - (void)animationDidStop:(CAAnimation *)animation finished:(BOOL)flag {
 
 #pragma mark - Setters
 
+- (void)setElevation:(MDCShadowElevation)elevation {
+  if (elevation == _elevation) {
```
Comment (romoore) [inline components/BottomAppBar/src/MDCBottomAppBarView.m]: Please use `MDCCGFloatEqual()`. Two floats that are very close to one another (but not exactly the same) wouldn't warrant invoking the block below.

File: components/BottomAppBar/tests/unit/BottomAppBarTests.m [hunk 5]
```diff
@@ -202,4 +202,39 @@ - (void)testTraitCollectionDidChangeBlockCalledWithExpectedParameters {
   XCTAssertEqual(passedBottomAppBar, bottomAppBar);
 }
 
+- (void)testElevationDidChangeBlockCalledWhenElevationChangesValue {
```
Comment (romoore) [inline components/BottomAppBar/tests/unit/BottomAppBarTests.m:205]: Please add a test that the default value of `mdc_baseElevation` is negative.

**Review by romoore — CHANGES_REQUESTED** [review]
Requesting changes because calling setters with side-effects in `init` is potentially dangerous.

**Review by romoore — APPROVED** [review]
